src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+	m_current_iteration = 0;
+	m_complete = false;
+
+	SG_ADD(
+	    &m_current_iteration, "current_iteration",
+	    "Current Iteration of training", MS_NOT_AVAILABLE);
+	SG_ADD(
+	    &m_max_iterations, "max_iterations", "Maximum number of Iterations",
+	    MS_AVAILABLE);
+	SG_ADD(&m_complete, "complete", "Convergence status", MS_NOT_AVAILABLE);
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+	SG_UNREF(m_continue_features);
+}
+
+bool CIterativeLinearMachine::train_machine(CFeatures* data)
+{
+	m_current_iteration = 0;
+	m_complete = false;
```

Comment:
also reset the continue features here

Reply:
(and store the given features)

Reply:
we do not need to unref when we leave continue_train() right ?

Reply:
you need to ref when you store a reference, and unref when you forget it